src/envoy/http/authn/authn_store.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2018 Istio Authors. All Rights Reserved.
```

Comment:
Can we start with smt simpler, i.e without this config store. I think the caching should be later, per (authn) method. Also, I try to see if we can use the same proto in the API repo as is.  You can take a look at my attempt [here](https://github.com/diemtvu/proxy/tree/policy/src/envoy/http/authn).